### PR TITLE
do not double count

### PR DESCRIPTION
**Pull request by vvalecha519**
-Updated TVL by decreasing by double counted part before

**Can we make sure that this is fixed for the past as well so our graph does not look wonky**

### CODE REVIEW

**Comment by llamatester** [comment]
The adapter at projects/etherfi-liquid exports TVL: 
        
 
 
    ethereum                  538.30 M
    
    total                    538.30 M

**Comment by llamatester** [comment]
The adapter at projects/etherfi-lrt exports TVL: 
        
 
 
    ethereum                  698.03 M
    
    total                    698.03 M

**Comment by llamatester** [comment]
Error while running adapter at projects/etherfi-liquid: 
        
 
  
    
    
    Error: Data is outdated
    Truncated error stack:
    
        at tvl (/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/projects/etherfi-liquid/index.js:38:11)
        at async getTvl (/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/test.js:52:23)
        at async /home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/test.js:173:9
        at async Promise.all (index 0)
        at async Promise.all (index 1)
        at async /home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/test.js:216:3

**Comment by llamatester** [comment]
The adapter at projects/etherfi-liquid exports TVL: 
        
 
 
    ethereum                  539.62 M
    
    total                    539.62 M

**Comment by llamatester** [comment]
The adapter at projects/etherfi-lrt exports TVL: 
        
 
 
    ethereum                  1.05 B
    
    total                    1.05 B

File: projects/etherfi-lrt/index.js [hunk 1]
```diff
@@ -7,13 +9,43 @@ const vaults = [
...
+  '0x1D4F0F05e50312d3E7B65659Ef7d06aa74651e0C',
+]
+
+async function vaultsTvl(api) {
+  for (let i = 0; i < vaults.length; i++) {
+    const bv = vaults[i]
+    const ba = vaultAccountant[i]
+    const bvSupply = await api.call({
+      target: bv,
+      abi: 'uint256:totalSupply',
```
Comment (0xngmi) [inline projects/etherfi-lrt/index.js:26]: why call this on a different contract? why was previous code not working?

Comment (vvalecha519) [reply projects/etherfi-lrt/index.js:26]: there were multiple reason:
1) Some of the vaults don't directly hold the assets instead they own a contract that holds the assets so need to call totalSupply
2) some of the eth in weethk and weeths vaults were being doubled 
3) The totalSupply * quote -> value in the base asset we were not multipled by quote before (and the quote and totalSupply are from different contracts)

**Comment by g1nt0ki** [comment]
hi @vvalecha519 
I have refactored it a bit here: https://github.com/DefiLlama/DefiLlama-Adapters/pull/12277
But I havent merged it yet as this would lead to drop in overall tvl, this is because you are counting all ETH deposits in etherfi-lrt as EETH and this gets excluded from parent tvl, is that what you want?

**Comment by vvalecha519** [comment]
> hi @vvalecha519 I have refactored it a bit here: #12277 But I havent merged it yet as this would lead to drop in overall tvl, this is because you are counting all ETH deposits in etherfi-lrt as EETH and this gets excluded from parent tvl, is that what you want?

Yes, both WEETHK and WEETHS should be counted as eETH. Can you confirm before merging that the total TVL, including LRT, liquid, and staking, would be approximately 2,410,000 ETH? Additionally, can you confirm that the historical TVL will be recalculated using the new method so the graph won’t display a sudden drop, as there was no actual drop in TVL; the previous calculation was simply inaccurate?

**Comment by g1nt0ki** [comment]
yes.

**Pull request merged**